elif clang-min definition changes

milljm opened the pull request:
Build our minimum version of supported Clang

Closes #25991

Commit:
elif clang-min definition changes

Build our minimum version of supported Clang

Closes #25991

milljm commented:
Initial go... works manually. Need to create "allow_on_pr_num" test recipes...

loganharbour commented:
While we're here, we should put this in post:

```
export MOOSE_JOBS={{ MOOSE_JOBS or "1" }}
```

replace the `${MOOSE_JOBS:-6}` with just `${MOOSE_JOBS}` and add

```
{#- MOOSE_JOBS: Number of jobs to pass to the build                                           -#}
```

to the jinja comment at the top

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1870494/) on 676b18b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25992/site)

This comment will be updated on new commits.

Commit:
Changes necessary for proper MPICH build

Changes made to allow for MPICH to build properly with manual Clang
build.

Commit:
More Templates

Template the gcc-toolchain version. Maybe someday this will make
upgrading/downgrading easier.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1874759/) on c3c5580 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>bda7b7</th>
      <th colspan=4>#25992 c3c558</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.69%</td>
      <td>85.69%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>92782</td>
      <td>92778</td>
      <td>-4</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15490</td>
      <td>15494</td>
      <td>+4</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25992/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25992/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25992/coverage/xfem)

</details>



This comment will be updated on new commits.

milljm commented:
@loganharbour if you happen to be online, and think this is ready...

loganharbour commented:
Do we need anything else done since this is another full rebuild?

milljm commented:
Probably this: https://github.com/idaholab/moose/pull/26023

cticenhour commented on apptainer/mpich.def:
```diff
@@ -115,45 +148,75 @@ EOF
       FC=gfortran \
       F77=gfortran \
-      F90='' \
+      F90=gfortan \
```
```suggestion
      F90=gfortran \
```

Found a typo

milljm replied:
wow, how did this pas!?

milljm commented:
Closing in favor of #26023
